### PR TITLE
FormInput component and addition to Sign In and Sign Up pages

### PR DESCRIPTION
**Pull request by naeder311**
created FormInput, FormRadioCheckbox, Components, Created SignInModal and SignUpModal

### CODE REVIEW

**Comment by naeder311** [comment]
I am not too sure how to do the leading and trailing icons, except for directly putting them into the placeholder.

File: src/components/form/FormInput.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <Fragment>
+      <label htmlFor={label}>{label}</label>
+      <Field
+        type={type}
+        id={name}
+        name={name}
+        placeholder={placeholder}
+        icon={icon}
+        className="my-3 h-[44px] w-full rounded-lg pl-2 bg-neutral-50 border-neutral-400 border-[1px] drop-shadow-lg active:border-blue-500 focus:border-blue-500 disabled:bg-neutral-100 disabled:text-neutral-400 invalid:border-brown-500 text-neutral-900 required"
+      />
```
Comment (GeorgeIpsum) [inline src/components/form/FormInput.tsx:30]:
```
<div className="absolute top-0 h-[44px] left-[]">
{ icon }
</div>
```

File: src/components/form/FormInput.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  value,
+}: {
+  label?: string;
+  name: string;
+  placeholder: string;
+  type: string;
+  value?: string;
+  icon?: object;
+}) => {
+  return (
```
Comment (GeorgeIpsum) [inline src/components/form/FormInput.tsx:20]: additional props:
- overwriteClassName
- className -> ```const inputClasses = overrwriteClassName ? className : `my-3 h-[44px] w-full rounded-lg pl-2 bg-neutral-50 border-neutral-400 border-[1px] drop-shadow-lg active:border-blue-500 focus:border-blue-500 disabled:bg-neutral-100 disabled:text-neutral-400 invalid:border-brown-500 text-neutral-900 required ${className}\`;``` (also determine whatever padding is needed internally to account for icons)

File: src/components/form/FormInput.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { Field } from 'formik';
+import { Fragment } from 'react';
+
+const FormInput = ({
```
Comment (GeorgeIpsum) [inline src/components/form/FormInput.tsx]: define an interface for Props, should extend FieldProps (which should exist in formik library)

File: src/components/form/FormInput.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  label,
+  name,
+  placeholder,
+  type,
+  icon,
+  value,
```
Comment (GeorgeIpsum) [inline src/components/form/FormInput.tsx]: you can add a ...props spread here now that props extends FieldProps

File: src/components/form/FormInput.tsx (hunk 1 above)
Comment (GeorgeIpsum) [inline src/components/form/FormInput.tsx:30]: respread props here aka {...props}

File: src/components/form/FormInput.tsx (hunk 2 above)
Comment (GeorgeIpsum) [reply src/components/form/FormInput.tsx:20]: error strings

**Review by GeorgeIpsum — CHANGES_REQUESTED** [review]
very epic

File: src/components/form/FormInput.tsx (hunk 3 above)
Comment (naeder311) [reply src/components/form/FormInput.tsx:4]: FieldProps extends: field, form, and meta.  Does provide any value?

**Review by naeder311 — COMMENTED** [review]
created overwrite class

File: src/components/form/FormInput.tsx (hunk 3 above)
Comment (naeder311) [reply src/components/form/FormInput.tsx]: Having an issue with FieldProps.  Unable to make them conditionally and they are causing issues when importing data

**Review by naeder311 — COMMENTED** [review]
Update FormInput conditional icons, conditional className

**Review by GeorgeIpsum — APPROVED** [review]
Very cool

File: src/components/form/FormInput.tsx (hunk 1 above)
Comment (GeorgeIpsum) [reply src/components/form/FormInput.tsx:30]: respreading props requires cherry picking certain form input props. this experience sucks. we'll deal with it later

**Review by GeorgeIpsum — APPROVED** [review]

**Review by GeorgeIpsum — APPROVED** [review]

**Pull request merged**